text/0000-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,131 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Categories for crates are now deprecated and implicitly added as keywords instead. A new set of policies is added to allow the crates.io team to curate the way keywords are presented, replacing features such as the "Popular Categories" list on crates.io with a "Popular Keywords" instead.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Cargo supports two types of metadata when tagging the purpose/scope of a crate:
+
+1. Categories, which are restricted to a curated set of allowed categories.
+2. Keywords, which are unrestricted and user-defined.
+
+However, the reality is that the actual curation of categories is extremely loose, and the only process for adding a category is to open a PR for the crates.io repository. The required review is effectively asking for an "LGTM" reply from one of the crates.io team members, as seen in [the history for the `categories.toml` file on the repository](https://github.com/rust-lang/crates.io/commits/main/src/boot/categories.toml).
+
+Ultimately, the most important distinction between categories and keywords has really been that the crates.io team has control over the presentation of categories, although a more important distinction is that categories cannot be removed, only added.
+
+Recently, [a discussion](https://github.com/rust-lang/crates.io/discussions/6762) was opened in the crates.io repository on whether the cryptocurrencies category should be removed from crates.io, due to the plethora of issues surrounding them. This should not be treated as a reason for adopting the RFC (although it was a motivation to write it), but instead as something that brought up the fact that categories cannot be removed by policy.
+
+Because categories cannot be removed, they also cannot be renamed or otherwise curated by the community. However, by switching to keywords, we can effectively solve this problem; community members can simply start publishing their crates under different keywords and older, unsupported crates wouldn't have any issue remaining published under their older versions.
```

Comment:
@kornelski [commented earlier](https://github.com/rust-lang/rfcs/pull/3488#issuecomment-1721208225)

> Categories can be removed and renamed. crates.io could ignore existence of slugs it doesn't want, and create aliases for category slugs it wants to rename. On lib.rs I've completely deleted external-ffi-bindings, and renamed math to science::math.

(moved here to avoid it getting lost in the noise)

Reply:
Right, although crates.io still ultimately has to accept categories that were valid in the past as valid in the future, meaning they're not really removed. Like, I don't think it's very good UX to just passively ignore a category when uploading to crates.io, since the user added that for a reason and we shouldn't just get rid of it. And if we actually stopped builds because of it, that'd be breaking backwards compatibility.

Reply:
More so my point is that this motivation doesn't seem valid.  The RFC doesn't address the alternative that @kornelski mentions: of having category merges, renames, and removals through a normalization process.

> Like, I don't think it's very good UX to just passively ignore a category when uploading to crates.io, since the user added that for a reason and we shouldn't just get rid of it

That doesn't mean it can't be done but might be a downside to this alternative.  That is something for the approving team to weigh in on with seeing your RFC and not for you to evaluate and leave out.  Personally, I would be in favor of such an approach though it isn't my call.

Reply:
I tend to agree with @kornelski and @epage. While the crates.io team has never deleted a category so far, it is technically possible with some effort to do so.

```suggestion
```

Probably easiest to just remove this part :)